src/webpack/patchWebpack.ts

```diff
@@ -1,362 +1,507 @@
-// Thus, we use this hack to immediately access their wreq.m and patch all already existing factories
-Object.defineProperty(Function.prototype, "m", {
-    configurable: true,
-
-    set(v: any) {
-        Object.defineProperty(this, "m", {
-            value: v,
-            configurable: true,
-            enumerable: true,
-            writable: true
-        });
+// If this is the main Webpack, we also set up the internal references to WebpackRequire.
+define(Function.prototype, "m", {
+    enumerable: false,
+
+    set(this: AnyWebpackRequire, originalModules: AnyWebpackRequire["m"]) {
+        define(this, "m", { value: originalModules });
 
-        // When using react devtools or other extensions, we may also catch their webpack here.
-        // This ensures we actually got the right one
+        // Ensure this is one of Discord main WebpackInstances.
+        // We may catch Discord bundled libs, React Devtools or other extensions WebpackInstances here.
         const { stack } = new Error();
-        if (!(stack?.includes("discord.com") || stack?.includes("discordapp.com")) || Array.isArray(v)) {
+        if (!stack?.includes("http") || stack.match(/at \d+? \(/) || !String(this).includes("exports:{}")) {
             return;
         }
 
-        const fileName = stack.match(/\/assets\/(.+?\.js)/)?.[1] ?? "";
-        logger.info("Found Webpack module factory", fileName);
+        const fileName = stack.match(/\/assets\/(.+?\.js)/)?.[1];
+        logger.info("Found Webpack module factories" + interpolateIfDefined` in ${fileName}`);
 
-        patchFactories(v);
+        allWebpackInstances.add(this);
 
-        // Define a setter for the bundlePath property of WebpackRequire. Only the main Webpack has this property.
+        // Define a setter for the ensureChunk property of WebpackRequire. Only the main Webpack (which is the only that includes chunk loading) has this property.
         // So if the setter is called, this means we can initialize the internal references to WebpackRequire.
-        Object.defineProperty(this, "p", {
-            configurable: true,
-
-            set(this: WebpackInstance, bundlePath: string) {
-                Object.defineProperty(this, "p", {
-                    value: bundlePath,
-                    configurable: true,
-                    enumerable: true,
-                    writable: true
-                });
+        define(this, "e", {
+            enumerable: false,
 
+            set(this: WebpackRequire, ensureChunk: WebpackRequire["e"]) {
+                define(this, "e", { value: ensureChunk });
                 clearTimeout(setterTimeout);
-                if (bundlePath !== "/assets/") return;
 
-                logger.info(`Main Webpack found in ${fileName}, initializing internal references to WebpackRequire`);
+                logger.info("Main WebpackInstance found" + interpolateIfDefined` in ${fileName}` + ", initializing internal references to WebpackRequire");
```

Comment:
Whats the point of this change? It just makes it less intuitive what is going on, and it mixes formatted and concatting strings
```suggestion
                logger.info(`Main WebpackInstance found ${interpolateIfDefined} in ${fileName}, initializing internal references to WebpackRequire`);
```
I *think* thats the same thing? again, its really hard to read what that string is.

Reply:
interpolateIfDefined returns an empty string if any one of the variables used inside of it is nullish, so that's not quite the same